serverless.yaml

```diff
@@ -519,7 +524,16 @@ resources:
                     Resource:
                       - !GetAtt DynamodbKMSKey.Arn
                       - !GetAtt ElasticSearchKMSKey.Arn
+      DdbToEsDLQ:
+        Type: AWS::SQS::Queue
+        Properties:
+          MessageRetentionPeriod: 1209600 # 14 days in seconds 
   - extensions:
+      DdbToEsEventSourceMappingDynamodbResourceDynamoDBTableV2:
+        Properties:
+          DestinationConfig:
+            OnFailure:
+              Destination: !GetAtt DdbToEsDLQ.Arn

```

Comment:
Not strictly needed but consider explicitly adding `"MaximumRetryAttempts": 3` here.

Looks like serverless adds that by default, which is a bit weird since the AWS default for streams is to retry an unlimited number of times(until the message expires). 

At first glance this looks like a config error since it's hard to tell that serverless adds `MaximumRetryAttempts`. I think that having `OnFailure` without setting `MaximumRetryAttempts` is effectively useless since Lambda keeps retrying and never "fails", I'm not sure if that config combination is even allowed.

Reply:
So it is actually being set in the event block in the start: https://github.com/awslabs/fhir-works-on-aws-deployment/blob/06d8db534d0b993d7958f32fcda9a52ff5a4dc60/serverless.yaml#L119

would you suggest adding a comment here instead?

Reply:
nvm, we can leave it as is. 
I didn't realize the stream config was above. A comment may help but I think the Serverless framework `extensions` will aways have the issue that config ends up slit into 2 places.